src/inc/volatile.h

```diff
@@ -106,26 +106,43 @@
 #define VOLATILE_MEMORY_BARRIER()
 #endif // __GNUC__
 
+template<typename T>
+struct RemoveVolatile
+{
+   typedef T type;
+};
+
+template<typename T>
+struct RemoveVolatile<volatile T>
+{
+   typedef T type;
+};
+
+
 //
 // VolatileLoad loads a T from a pointer to T.  It is guaranteed that this load will not be optimized
 // away by the compiler, and that any operation that occurs after this load, in program order, will
 // not be moved before this load.  In general it is not guaranteed that the load will be atomic, though
 // this is the case for most aligned scalar data types.  If you need atomic loads or stores, you need
 // to consult the compiler and CPU manuals to find which circumstances allow atomicity.
 //
+// Starting at version 3.8, clang errors out on initializing of type int * to volatile int *. To fix this, we add two templates to cast away volatility
```

Comment:
@cydhaselton what do you mean by "different areas"? It is possible to build coreclr with different versions of the clang toolchain. For official builds and so also for CI lab testing for ARM64, the clang 3.7 was chosen. There is no reason why we could not use higher versions of the compiler. We still use clang 3.5 for x64 builds and we use clang 3.9 for ARM32. We would like to unify the compiler version soon.